email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,393 @@
+        Properties prop = extractProperties(config);
+        
+        try {
+            session = Session.getDefaultInstance(prop);
+            store = session.getStore(mailStoreType);
+            store.connect(host, user, password);
+        } catch (MessagingException e) {
+            logger.error("Failed to start email session");
+        }
+    }
+    
+    /**
+     * Extract properties from configuration object
+     * 
+     * @param config  Configuration object that stores mail server information, username, and password
+     * @return the properties file
+     */
+    private Properties extractProperties(Config config) {
+        Properties prop = new Properties();
+        for (java.util.Map.Entry<java.lang.String, ConfigValue> entry : config.entrySet()) {
+            prop.setProperty(entry.getKey(), config.getString(entry.getKey()));
+        }        
+        return prop;
+    }
+    /**
+     * Retrieve folders and create them if they do not already exist
+     */
+    public void initFolders() {
+        try {
+            if (!store.getFolder(EmailType.ERROR.getValue()).exists()) {
+                store.getFolder(EmailType.ERROR.getValue()).create(HOLDS_MESSAGES);
+            }
+            if (!store.getFolder(EmailType.COMPLETE.getValue()).exists()) {
+                store.getFolder(EmailType.COMPLETE.getValue()).create(HOLDS_MESSAGES);
+            }
+            errorFolder = store.getFolder(EmailType.ERROR.getValue());
+            completeFolder = store.getFolder(EmailType.COMPLETE.getValue());
+            inboxFolder = store.getFolder(EmailType.INBOX.getValue());
+        } catch (MessagingException e) {
+            logger.error("Failed to initialize folders");
+        }
+    }
+
+    /**
+     * Determine which folder should be accessed
+     * 
+     * @param folder   the folder to be returned. Expressed by an integer value
+     * @return Folder   the folder to be accessed
+     */
+    public Folder determineFolder(EmailType folder) {
+        switch (folder) {
+            case INBOX:
+                return inboxFolder;
+            case ERROR:
+                return errorFolder;
+            case COMPLETE:
+                return completeFolder;
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Determine which message array should be used
+     * 
+     * @param msg   the message to be accessed. Expressed by an integer value
+     * @return Message   the message to be used
+     */
+    public Message[] determineMsg(EmailType msg) {
+        switch (msg) {
+            case UNREAD:
+                return unreadMessages;
+            case READ:
+                return processingMessages;
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Open specified folder
+     * 
+     * @param folder   the folder to be opened. Expressed by an integer value  
+     */
+    public void openFolder(EmailType folder) {
```

Comment:
Are you ever you going to use the EmailManager and not open all the folders? How about having it open all the folders in the constructor (in the initFolders() method)?